parsons/empower/empower.py

```diff
@@ -0,0 +1,261 @@
+from parsons.utilities.api_connector import APIConnector
+from parsons.utilities import check_env
+from parsons.etl import Table
+import logging
+from datetime import datetime
+
+logger = logging.getLogger(__name__)
+
+EMPOWER_API_ENDPOINT = "https://api.getempower.com/v1/export"
+
+
+class Empower(object):
+    """
+    Instantiate class.
+
+        `Args:`
+            api_key: str
+                The Empower provided API key.The Empower provided Client UUID. Not
+                required if ``EMPOWER_API_KEY`` env variable set.
+            empower_uri: str
+                The URI to access the Empower API. The default is currently set to
+                https://api.getempower.com/v1/export. You can set an ``EMPOWER_URI`` env
+                variable or use this URI parameter if a different endpoint is necessary.
+            cache: boolean
+                The Empower API returns all account data after each call. Setting cache
+                to ``True`` stores the blob and then extracts Parsons tables for each method.
+                Setting cache to ``False`` will download all account data for each method call.
+    """
+
+    def __init__(self, api_key=None, empower_uri=None, cache=True):
+        self.api_key = check_env.check("EMPOWER_API_KEY", api_key)
+        self.empower_uri = (
+            check_env.check("EMPOWER_URI", empower_uri, optional=True) or EMPOWER_API_ENDPOINT
+        )
+        self.headers = {"accept": "application/json", "secret-token": self.api_key}
+        self.client = APIConnector(
+            self.empower_uri,
+            headers=self.headers,
+        )
+        self.data = None
+        self.data = self._get_data(cache)
+
+    def _get_data(self, cache):
+        """
+        Gets fresh data from Empower API based on cache setting.
+        """
+
```

Comment:
Oh if self.data was set to None so that this method works, it's probably better to move the logic inside the method. You could also do something like `self.get("data", None)` where it will look for data and return None if nothing's there instead of raising a KeyError